lnrpc/invoicesrpc/invoices_server.go

```diff
@@ -335,9 +335,26 @@ func (s *Server) AddHoldInvoice(ctx context.Context,
 		GetAlias:              s.cfg.GetAlias,
 	}
 
-	hash, err := lntypes.MakeHash(invoice.Hash)
-	if err != nil {
-		return nil, err
+	var preimage *lntypes.Preimage
+	var hash *lntypes.Hash
+	var err error
+
+	if len(invoice.Hash) == 0 {
+		preimage, err = lntypes.RandomPreimage()
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		hash0, err := lntypes.MakeHash(invoice.Hash)
```

Comment:
i can ofc rename the variable perhaps `hash_pointer`, but this is necessary as a workaround `var hash *lntypes.Hash` to be a pointer to some value or nil. this is to make a value and make it a pointer. 

my first time writing in golang so maybe this is not the best way... what do you suggest otherwise?